Dockerfile

```diff
@@ -71,31 +59,33 @@ FROM builder as workflow-controller-build
 RUN cat .dockerignore >> .gitignore
 RUN git status --porcelain | cut -c4- | xargs git update-index --skip-worktree
 
-RUN --mount=type=cache,target=/root/.cache/go-build make dist/workflow-controller
+RUN --mount=type=cache,target=/go/pkg/mod --mount=type=cache,target=/root/.cache/go-build make dist/workflow-controller
 
 ####################################################################################################
 
 FROM builder as argocli-build
 
 RUN mkdir -p ui/dist
 COPY --from=argo-ui ui/dist/app ui/dist/app
-# stop make from trying to re-build this without yarn installed
-RUN touch ui/dist/node_modules.marker
-RUN touch ui/dist/app/index.html
 
 # Tell git to forget about all of the files that were not included because of .dockerignore in order to ensure that
 # the git state is "clean" even though said .dockerignore files are not present
 RUN cat .dockerignore >> .gitignore
 RUN git status --porcelain | cut -c4- | xargs git update-index --skip-worktree
 
-RUN --mount=type=cache,target=/root/.cache/go-build make dist/argo
+RUN --mount=type=cache,target=/go/pkg/mod --mount=type=cache,target=/root/.cache/go-build make dist/argo
+
+####################################################################################################
+
+# NOTE: kubectl version should be one minor version less than https://storage.googleapis.com/kubernetes-release/release/stable.txt
+FROM bitnami/kubectl:1.24.8 as kubectl
```

Comment:
do we need this step, can't we curl the binary in the `builder` step and copy it later?
https://github.com/argoproj/argo-workflows/blob/1225d8b546c1e5093047d8e6e8a46d053f051d97/Dockerfile#L52-L53

https://github.com/argoproj/argo-workflows/blob/1225d8b546c1e5093047d8e6e8a46d053f051d97/Dockerfile#L97

Reply:
Using the bitnami base image is less verbose and keeps the builder step smaller. Additionally, if we want to upgrade the kubectl version, it'll only affect the argoexec image rather than all three images, which would be the case if we move the kubectl import to the builder step

Reply:
Also, using the bitnami import will prevent us from having to worry about the OS/arch when importing kubectl, which is another nice to have